libnmstate/schemas/operational-state.yaml

```diff
@@ -287,6 +287,23 @@ definitions:
                     type: string
                   access-tag:
                     type: string
+                link-aggregation:
```

Comment:
I do not understand the indentation. Is it supposed to be at the same level as the previous `access-tag` key?

But the main question at this level is: How will we differentiate between a regular port and a link-aggregation one? Please consider that nmstate likes partial editing in general.

Reply:
link-aggretation port has to have two or more interfaces attached. I don't see how that breaks the partial editation, could you elaborate?

Reply:
You use the existence of the subtree do determine if the user requested a bond or not. This will not work well with partial editation. If the user  wants to change a port vlan for example, he will not mention the subtree.. but then, you will not be able to differentiate that request from just making that port a non bond. I think we need here an ‘enable’ flag.

Reply:
That won't work the other way around though. In OVS database, the only mark that something is a bond is that multiple interfaces are under the same port. Nothing else, no attributes, nothing.

Reply:
Ah, now I get you. So after all we should have kept the `type` of port :)

Reply:
Any suggestion how it should look like? @EdDev

Reply:
We can still check if it is a bond or simple port while applying a partial config. No need for an additional attribute. While generating proxy interfaces, we can check from the system for the current number of ifaces. That is what I did in https://github.com/nmstate/nmstate/pull/514 if I remember correctly.

Reply:
I have an idea: we can declare that for changing a bond to a non bond, it needs to be “absent” first. That will work.
But we now need an absent property.

Reply:
> We can still check if it is a bond or simple port while applying a partial config. No need for an additional attribute. While generating proxy interfaces, we can check from the system for the current number of ifaces. That is what I did in #514 if I remember correctly.

The problem is not about what exists currently, but what the client intentions are.
But I think I answered it earlier now.

Reply:
I don't get what you are trying to solve. Changing bond to a non-bond can look like following:

```yaml
port:
- name: bond1
  link-aggregation:
    slaves:
      - name: eth0
      - name: eth1
```

To:

```yaml
port:
- name: eth0
```

You cannot partially change list of slaves. There is no such support in linux bonding nor linux bridge afaik.

Reply:
@EdDev please give me a specific example of a change that you think would not work with the current schema and I will tell you how would it be tackled or why it should not be supported.

Reply:
Your example should cause a failure and not change a state. If in practice tries to define a contradicting state: on one part eth0 is a port, in parallel it is a bond slave.

Reply:
Ah, you want a partial editing within the port? Oof...

Any idea how to tackle it (other than creating link-aggregation type and making nmstate into a circus)?

Reply:
> I have an idea: we can declare that for changing a bond to a non bond, it needs to be “absent” first. That will work.
> But we now need an absent property.

Reply:
> > I have an idea: we can declare that for changing a bond to a non bond, it needs to be “absent” first. That will work.
> > But we now need an absent property.

IMHO this violates the idea that Nmstate users declare what they want instead of how they get it. Also, where do you want to declare this?

Reply:
This is what we do with routes and rules, so I do not know what this violates.
I will open a discussion on the PR root on this with an example.